src/main/c/ssl.c

```diff
@@ -1366,59 +1476,83 @@ TCN_IMPLEMENT_CALL(jlong, SSL, makeNetworkBIO)(TCN_STDARGS,
     BIO *internal_bio;
     BIO *network_bio;
 
-    UNREFERENCED(o);
-
     if (ssl_ == NULL) {
         tcn_ThrowException(e, "ssl is null");
-        goto fail;
+        return 0;
     }
 
     if (BIO_new_bio_pair(&internal_bio, 0, &network_bio, 0) != 1) {
         tcn_ThrowException(e, "BIO_new_bio_pair failed");
-        goto fail;
+        return 0;
     }
 
+    UNREFERENCED(o);
+
     SSL_set_bio(ssl_, internal_bio, internal_bio);
 
     return P2J(network_bio);
- fail:
-    return 0;
 }
 
 // Free a BIO * (typically, the network BIO)
 TCN_IMPLEMENT_CALL(void, SSL, freeBIO)(TCN_STDARGS,
                                        jlong bio /* BIO * */) {
-    BIO *bio_;
+    BIO *bio_ = J2P(bio, BIO *);
+
+    if (bio_ == NULL) {
+        tcn_ThrowException(e, "bio is null");
```

Comment:
For `closeBIO` we didn't throw an exception but instead we just did the close if the pointer was not null. Should we do something similar here?